testcases/OpTestKexec.py

```diff
@@ -0,0 +1,129 @@
+#
+
+import os
+import pexpect
+import unittest
+import time
+import tempfile
+import OpTestLogger
+
+import OpTestConfiguration
+from common import OpTestInstallUtil
+from common.OpTestSystem import OpSystemState
+from common.Exceptions import CommandFailed
+from common.OpTestConstants import OpTestConstants as BMC_CONST
+from common.OpTestError import OpTestError
+
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+class OpTestKexec(unittest.TestCase):
+
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.cv_HOST = conf.host()
+        self.cv_BMC = conf.bmc()
+        self.bmc_type = conf.args.bmc_type
+        self.util = self.cv_SYSTEM.util
+        self.pdbg = conf.args.pdbg
+        self.basedir = conf.basedir
+        self.c = self.cv_SYSTEM.console
+        self.kernel_image = conf.args.kernel_image
+        self.initrd_image = conf.args.initrd_image
+        if self.bmc_type == "FSP_PHYP":
+            self.is_lpar = True
+            self.hmc_user = conf.args.hmc_username
+            self.hmc_password = conf.args.hmc_password
+            self.hmc_ip = conf.args.hmc_ip
+            self.lpar_name = conf.args.lpar_name
+            self.system_name = conf.args.system_name
+            self.cv_HMC = self.cv_SYSTEM.hmc
+            try: self.cpu_resource = conf.args.cpu_resource
+            except AttributeError:
+                self.cpu_resource = 1
+            try: self.mem_resource = conf.args.mem_resource
+            except AttributeError:
+                self.mem_resource = 2048
+        try: self.url = conf.args.url
+        except AttributeError: 
+            self.url = "http://liquidtelecom.dl.sourceforge.net/project/ebizzy/ebizzy/0.3/ebizzy-0.3.tar.gz"
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        res = self.c.run_command("cat /etc/os-release")
+        if "Ubuntu" in res[0] or "Ubuntu" in res[1]:
+            self.distro = "ubuntu"
+        elif 'Red Hat' in res[0] or 'Red Hat' in res[1]:
+            self.distro = 'rhel'
+        elif 'SLES' in res[0] or 'SLES' in res[1]:
+            self.distro = 'sles'
+        else:
+            raise self.skipTest("Test currently supported only on ubuntu, sles and rhel")
+
+    ##
+    # @brief This function will test the kexec load
+    #
+    def kexec_load(self):
+        try:
+            self.c.run_command("kexec -l /boot/{} --initrd /boot/{} --append=\"`cat /proc/cmdline`\"".format(self.kernel_image,self.initrd_image))
+        except CommandFailed:
+            raise OpTestError("kexec -l failed to load kernel")
+
+        kexec_loaded_status = self.c.run_command("cat /sys/kernel/kexec_loaded")
+        if kexec_loaded_status[0] == "1":
+            log.info("kernel loaded successfully with kexec_loaded_status = \"%s\"", kexec_loaded_status[0])
+        else:
+            log.info("New kernel not loaded- Failed with kexec_loaded_status = \"%s\"", kexec_loaded_status[0])
```

Comment:
better to check if it fails and log info ,,   succesfull load check is need not be logged

Reply:
Done